tests/variable_creation_test.py

```diff
@@ -245,6 +245,40 @@ def test_array_empty_dims():
                         sc.scalar(1.1, variance=1.1))
 
 
+def test_array_dim_from_keyword_fail_multiple():
+    with pytest.raises(ValueError):
+        sc.array(x=[4], y=[5])
+
+
+def test_array_dim_from_keyword_fail_redundant_dims():
+    with pytest.raises(ValueError):
+        sc.array(dims=['x'], x=[4])
+    with pytest.raises(ValueError):
+        sc.array(dims=['y'], x=[4])
+
+
+def test_array_dim_from_keyword_fail_redundant_values():
+    with pytest.raises(ValueError):
+        sc.array(x=[4], values=[4])
+
+
+def test_array_dim_from_keyword():
+    var = sc.array(xx=[2, 3, 4], unit='K', dtype='float32')
+    assert sc.identical(
+        var, sc.array(dims=['xx'], values=[2, 3, 4], unit='K', dtype='float32'))
+
+
+def test_array_dim_from_keyword_with_variances():
```

Comment:
Should this actually be allowed?
It is non-ambiguous but seems a bit odd.

Reply:
I also thought it was a bit unusual, but it seemed harmless and preventing it would require extra code, so I left it in. I don't have a definite opinion though.

Reply:
Fine by me